dbus/methods_test.go

```diff
@@ -739,9 +754,527 @@ func TestStartStopTransientUnit(t *testing.T) {
 	}
 }
 
+// Ensure that oneshot transient unit starting and stopping works.
+func TestStartStopTransientUnitOneshot(t *testing.T) {
```

Comment:
initial eyeballing suggests a lot of copy-paste between these test functions - have you tried to parameterise it into a test table?